code/game/objects/structures/door_assembly.dm

```diff
@@ -226,18 +209,7 @@
 	. = TRUE
 	if(!I.tool_use_check(user, 0))
 		return
-	if(mineral)
-		var/obj/item/stack/sheet/mineral/mineral_path = text2path("/obj/item/stack/sheet/mineral/[mineral]")
-		user.visible_message("<span class='notice'>[user] welds the [mineral] plating off [src].</span>",\
-			"<span class='notice'>You start to weld the [mineral] plating off [src]...</span>",\
-			"<span class='warning'>You hear welding.</span>")
-		if(!I.use_tool(src, user, 40, volume = I.tool_volume))
-			return
-		to_chat(user, "<span class='notice'>You weld the [mineral] plating off.</span>")
-		new mineral_path(loc, 2)
-		var/obj/structure/door_assembly/PA = new previous_assembly(loc)
```

Comment:
With this chunk of code going, it looks like `previous_assembly` is only set and never used meaningfully anymore. Likewise with the `previous_airlock` var on airlocks I believe.

Seems these were only used for remembering which type the assembly/airlock was before applying a mineral, but since we make them directly now, we don't need them. Could you remove references to these vars?

Everything else looks good.

Reply:
Ah right! Forgot bout that, my bad. Il fix it later today ^^

Reply:
Done! Tested everything one final time just to be safe.